ic-agent/src/agent/mod.rs

```diff
@@ -633,6 +663,50 @@ impl Agent {
         msg.extend_from_slice(IC_STATE_ROOT_DOMAIN_SEPARATOR);
         msg.extend_from_slice(&root_hash);
 
+        let contains_delegations = cert.delegation.is_some();
+        if disable_range_check && contains_delegations && !is_delegation {
+            let paths = cert.tree.list_paths();
+            let t: Label = "time".into();
+            let rs: Label = "request_status".into();
+            // rid contains <request_id> from paths /request_status/<request_id> if present in the certificate
+            let mut rid: Option<Label> = None;
+            for p in paths {
+                if !(p == vec![t.clone()] || p[0] == rs) {
+                    // the path has neither the form /time nor /request_status/*
+                    return Err(AgentError::CertificateVerificationFailed());
```

Comment:
Up until now, the `CertificateVerificationFailed` error referred to a specific cause: `bls::core_verify()` returned something other than `BLS_OK`, meaning the certificate signature didn't match the expected value.

I think it woiuld be helpful to someone diagnosing these errors if we introduced a new error type, `CertificateVerificationRejected(reason)`, where reason is an enumeration that gives more information about the reason for the rejection of the certificate.  What do you think?